hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go

```diff
@@ -2275,32 +2327,23 @@ func TestValidateReleaseImage(t *testing.T) {
 					},
 				},
 			},
-			expectedResult: nil,
+			releaseImageLoookup: nil, //won't be called
+			expectedResult:      nil,
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			g := NewGomegaWithT(t)
+			mockedProviderWithOpenShiftImageRegistryOverrides := releaseinfo.NewMockProviderWithOpenShiftImageRegistryOverrides(mockCtrl)
+			mockedProviderWithOpenShiftImageRegistryOverrides.EXPECT().Lookup(gomock.Any(), gomock.Any(), gomock.Any()).
+				DoAndReturn(tc.releaseImageLoookup).AnyTimes()
 			r := &HostedClusterReconciler{
 				CertRotationScale: 24 * time.Hour,
 				Client:            fake.NewClientBuilder().WithObjects(tc.other...).Build(),
-				ReconcileMetadataProviders: func(ctx context.Context, imgOverrides map[string]string) (releaseinfo.ProviderWithOpenShiftImageRegistryOverrides, hyperutil.ImageMetadataProvider, error) {
-					return &fakereleaseprovider.FakeReleaseProvider{
-							ImageVersion: map[string]string{
-								"image-4.7.0":  "4.7.0",
-								"image-4.10.0": "4.10.0",
-								"image-4.11.0": "4.11.0",
-								"image-4.12.0": "4.12.0",
-								"image-4.12.1": "4.12.1",
-								"image-4.13.0": "4.13.0",
-								"image-4.14.0": "4.14.0",
-								"image-4.15.0": "4.15.0",
-								"image-4.16.0": "4.16.0",
-								"image-4.17.0": "4.17.0",
-								"image-4.18.0": "4.18.0",
-							},
-						},
```

Comment:
To understand it better, how this maps with the versions removed in the test, to test the same as we have now with mockgen? (sorry I'm not an expert on mockgen)

Reply:
For what I understand the test won't need that images. Maybe at the beginning it was the case or this come from some cut and paste.  Again, this is a good reason to have mock generated and in the *_test.go file only.